### DISCUSSION TITLE
Handling wagmi-connector versioning and cross-version compatibility

### DISCUSSION THREAD

**Discussion by theodormarcu** [wevm/wagmi#1259] [Q&A]
Hi! What's the recommended way to version third-party wagmi-connectors? We want to support multiple versions of `wagmi` for our [wallet wagmi connector](https://www.npmjs.com/package/@marblexyz/wagmi-connector) (e.g. `0.6.x`, `0.7.x`), and we were thinking about matching our versioning to wagmi's (e.g. our version is `0.6.x` for wagmi's `0.6.x`). Another idea is adding a prefix or suffix (e.g. `wagmi0.6-X.Y.Z`, `wagmi0.7-X.Y.Z`) where X.Y.Z is our version.

How are other third party connectors handling this? Any suggestions would be welcome!

**Comment by tmm** [comment]
Matching the version could work, but is probably more trouble than it is worth for a few reasons:

- In order to match the version, you might need to release a new version (since wagmi did) even though nothing has changed for your connector.
- You might ship a bug in your connector, which could require a new version and put the two out of sync anyway.
- You need to choose what version of wagmi to track. There are currently two packages, Vanilla JS `@wagmi/core` and React `wagmi` (maybe more in the future, like `@wagmi/vue`). They tend to follow similar version bumps, but it is not always the case.

Instead, what I recommend is adding a [version range](https://semver.npmjs.com) and updating that for new `wagmi`/`@wagmi/core` releases.

**Reply by theodormarcu** [reply]
Thanks for the reply! Would it be possible to share an example with me?

Say my current version of `@marblexyz/wagmi-connector` supports `0.6.x` and I also want to release a version that supports `0.7.x` while being able to make changes to my previous version (i.e. release patches). How would you version the connector in this case for each version?

**Reply by theodormarcu** [reply]
cc @tmm sorry to ping, but would love an answer here!
